bitchat/Features/media/ImageUtils.swift

```diff
@@ -0,0 +1,149 @@
+import Foundation
+#if os(iOS)
+import UIKit
+#else
+import AppKit
+import ImageIO
+import UniformTypeIdentifiers
+#endif
+
+enum ImageUtilsError: Error {
+    case invalidImage
+    case encodingFailed
+}
+
+enum ImageUtils {
+    private static let compressionQuality: CGFloat = 0.85
+    private static let targetImageBytes: Int = 60_000
+
+    static func processImage(at url: URL, maxDimension: CGFloat = 512) throws -> URL {
+        let data = try Data(contentsOf: url)
+        #if os(iOS)
+        guard let image = UIImage(data: data) else { throw ImageUtilsError.invalidImage }
+        return try processImage(image, maxDimension: maxDimension)
+        #else
+        guard let image = NSImage(data: data) else { throw ImageUtilsError.invalidImage }
+        return try processImage(image, maxDimension: maxDimension)
+        #endif
+    }
+
+    #if os(iOS)
+    static func processImage(_ image: UIImage, maxDimension: CGFloat = 512) throws -> URL {
+        let scaled = scaledImage(image, maxDimension: maxDimension)
+        var quality = compressionQuality
+        guard var jpegData = scaled.jpegData(compressionQuality: quality) else {
```

Comment:
we might want to check if the image is already small enough to avoid unnecessary processing